### ISSUE TITLE
to_yaml converts entire template, including json values

### ISSUE THREAD

**Issue by daschi** [awslabs/aws-cfn-template-flip#75]
Hello and thank you so much for this package! 

I was wondering if anyone had come across the issue of trying to convert a json template to yaml that includes some JSON values. For example, the [`AWS::StepFunctions::StateMachine`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-stepfunctions-statemachine.html#cfn-stepfunctions-statemachine-definitionstring) resource has a property `DefinitionString` which requires the [Amazon States Language JSON-based syntax](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-amazon-states-language.html). Currently the converter will convert the value of DefinitionString from JSON to yaml format, which is invalid syntax for that CloudFormation resource. 

I was considering a potential solution would be to convert the entire template to yaml, and then load the yaml file as an object and iterate each of the CloudFormation `Resources` and use the `to_json` method on the values that should remain JSON (like the DefinitionString for `AWS::StepFunctions::StateMachine`). Would love to hear of any other solutions you've used or thought of or if you have any feedback on this approach! Thanks!

**Comment by stilvoid** [comment]
Hi @daschi and thanks for the bug report. Do you have an example template that I could work from to see if I can find a nice way around this?

**Comment by koiker** [comment]
It's a interesting topic as normally when I build a YAML template some payloads/values are acceptable in yaml or json and are interpreted correctly by Cloudformation.
A sample template with the issue will be good to know how to handle the parser and maybe add some pre-processor function to keep some data as-is.

**Comment by daschi** [comment]
@stilvoid Here's an example json template, the yaml output, and then the valid yaml template that AWS StateMachine resource expects:

```json
# input json template
{
    "AWSTemplateFormatVersion": "2010-09-09",
    "Transform": "AWS::Serverless-2016-10-31",
    "Resources": {
        "ExampleStateMachineResource": {
            "Type": "AWS::StepFunctions::StateMachine",
            "Properties": {
                "DefinitionString": {
                  "StartAt": "First State",
                  "States": {
                    "First State": {
                      "Type": "Task",
                      "Resource": "FunctionARN",
                      "Next": "Second State"
                    },
                    "Second State": {
                      "Type": "Task",
                      "Resource": "FunctionARN",
                      "End": true
                    }
                  }
                }
            }
        }
    }
} 
```

```yaml
# output yaml template 
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Resources:
  DistrictNumStudents:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      DefinitionString:
        StartAt: First State
        States:
          First State:
            Type: Task
            Resource: FunctionARN
            Next: Second State
          Second State:
            Type: Task
            Resource: FunctionARN
            End: true
```

```yaml
# Valid yaml template for StateMachine resource
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Resources:
  DistrictNumStudents:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      DefinitionString: |-
        {
          "StartAt": "First State",
          "States": {
            "First State": {
              "Type": "Task",
              "Resource": "FunctionARN",
              "Next": "Second State"
            },
            "Second State": {
              "Type": "Task",
              "Resource": "FunctionARN",
              "End": true
            }
          }
        }
```

Let me know if I can help with any more examples!

**Comment by koiker** [comment]
I got the problem and will try to add some configuration to the tool to keep parts of the value as-is based on the resource type.

**Comment by roganov** [comment]
@daschi `DefinitionString` must be a string according to docs, not a JSON document.

**Comment by koiker** [comment]
I'm working in this issue and will get back soon with the feature.

**Comment by koiker** [comment]
Fixed and send PR: https://github.com/awslabs/aws-cfn-template-flip/pull/80

**Comment by stilvoid** [comment]
Fixed in #80
